### ISSUE TITLE
Fix false positives for single-line comments in selector-descendant-combinator-no-non-space

### ISSUE THREAD

**Issue by Stanja219** [stylelint/stylelint#3975]
> Clearly describe the bug

Lots of comments are marked with a 
selector-descendant-combinator-no-non-space
error

> Which rule, if any, is the bug related to?

selector-descendant-combinator-no-non-space

> What CSS is needed to reproduce the bug?

```css
//* ==================================================
//   HTML5 display definitions
//   =================================================//*/

or

//**
//* Correct `block` display not defined in IE 8/9.
//*/

html,
button, // inline Comment
input,
select,
textarea {
    color: red
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
    "ignoreFiles": "*/vendors/**/*",
    "rules": {
        "block-no-empty": true,
        "block-closing-brace-empty-line-before": "never",
        "block-opening-brace-space-before": "always",

        "color-hex-case": "lower",
        "color-hex-length": "short",
        "color-no-invalid-hex": true,
        "color-named": "never",
        
        "declaration-block-semicolon-newline-after": "always",
        "declaration-block-trailing-semicolon": "always",
        "declaration-colon-space-after": "always",
        "declaration-no-important": true,

        "font-weight-notation": "numeric",

        "function-comma-space-after": "always-single-line",
        "function-comma-space-before": "never",
        "function-whitespace-after": "always",

        "indentation": [4, {
            "except": ["value"]
        }],

        "length-zero-no-unit": true,

        "max-empty-lines": 2,
        "max-nesting-depth": [
            5,{
                "ignore": ["blockless-at-rules"]
            }
        ],

        "media-feature-colon-space-after": "always",
        "media-feature-colon-space-before": "never",
        "media-feature-name-case": "lower",
        "media-feature-parentheses-space-inside": "never",
        "media-feature-range-operator-space-after": "always",
        "media-feature-range-operator-space-before": "always",

        "no-duplicate-selectors": true,
        "no-eol-whitespace": [true, {
            "ignore": ["empty-lines"]
        }],
        "no-missing-end-of-source-newline": true,

        "number-leading-zero": "always",

        "property-no-unknown": true,
        "property-case": "lower",

        "rule-empty-line-before": [ "always", {
            "except": ["first-nested"],
            "ignore": ["after-comment"]
        } ],

        "selector-attribute-brackets-space-inside": "never",
        "selector-attribute-operator-space-after": "never",
        "selector-attribute-operator-space-before": "never",
        "selector-attribute-quotes": "always",
        "selector-descendant-combinator-no-non-space": true,
        "selector-combinator-space-after": "always",
        "selector-combinator-space-before": "always",
        "selector-list-comma-newline-after": "always",

        "selector-pseudo-class-case": "lower",
        "selector-pseudo-class-parentheses-space-inside": "never",
        "selector-pseudo-element-colon-notation": "double",
        "selector-type-case": "lower",

       
        "selector-pseudo-class-no-unknown": true,
        "selector-type-no-unknown": true,

        "string-quotes": "double",

        "unit-case": "lower",
        "unit-no-unknown": true,

        "value-keyword-case": "lower",

        "value-list-comma-space-after": "always-single-line",
        "value-list-comma-space-before": "never"
    }
}

```

> Which version of stylelint are you using?

I'm using stylelint version 9.9.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

gulp-stylelint with stylelint-report.txt
and VSCode

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

less

> What did you expect to happen?

"No warnings to be flagged."

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
Unexpected " "                                                  selector-descendant-combinator-no-non-space
```

**Comment by jeddy3** [comment]
@Stanja219 Thanks for the report and for using the template.

I can reproduce this using the online demo.

I'm going to label as "help wanted". It could be an issue with the rule code itself or one with the upstream Less parser we use (postcss-less). If you'd like to help out, please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by jeddy3** [comment]
Closing [as stylistic rules are frozen](https://github.com/stylelint/stylelint/issues/5674). The community is welcome to migrate the rule [to a plugin](https://stylelint.io/developer-guide/plugins) and fix this bug.
